### PR TITLE
Feature/create task repository

### PR DESCRIPTION
**Pull request by LucyFreelanceDeveloper**

### CODE REVIEW

**Review by GeorgeFreelanceDeveloper — COMMENTED** [review]
Added comments

File: src/main/java/org/samples/todos/repository/TaskRepository.java [hunk 1]
```diff
@@ -1,23 +1,46 @@
...
 
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.ArrayList;
 import java.util.List;
 
 public class TaskRepository {
 
+    private static String TASKS_FILE_NAME = "src/main/resources/tasks.json";
```
[the remark this answers is not in the document]
Comment (LucyQuantAnalyst) [reply src/main/java/org/samples/todos/repository/TaskRepository.java:15]: Done

File: src/test/java/org/samples/todos/repository/TaskRepositoryTest.java [hunk 2]
```diff
@@ -29,27 +34,25 @@ public void testLoad() throws ParseException {
...
 
-        ClassLoader classLoader = getClass().getClassLoader();
-        File file = new File(classLoader.getResource(TASKS_FILE_NAME_FOR_SAVE).getFile());
-        actualResult = FileUtils.readFileToString(file, "UTF-8");
+        try (InputStream inputStream = Files.newInputStream(Paths.get(TASKS_FILE_NAME_FOR_SAVE))) {
+            actualResult = new String(inputStream.readAllBytes(), StandardCharsets.UTF_8);
+        }
 
-        assertEquals(expectedResult, actualResult);
+        assertEquals(expectedResult.replaceAll("\\s", ""), actualResult.replaceAll("\\s", ""));
```
[the remark this answers is not in the document]
Comment (LucyQuantAnalyst) [reply src/test/java/org/samples/todos/repository/TaskRepositoryTest.java:48]: Done

File: src/main/java/org/samples/todos/repository/TaskRepository.java [hunk 3]
```diff
@@ -1,23 +1,46 @@
...
 
     public void save(List<TaskGroup> taskGroups) {
-        //TODO: @Lucka please implement me
-        // Example: use jackson library https://www.baeldung.com/jackson-object-mapper-tutorial
+        ObjectMapper objectMapper = new ObjectMapper();
+
+        try {
+            objectMapper.writeValue(new File(TASKS_FILE_NAME), taskGroups);
+        } catch (IOException e) {
+            e.printStackTrace();
```
[the remark this answers is not in the document]
Comment (LucyQuantAnalyst) [reply src/main/java/org/samples/todos/repository/TaskRepository.java:42]: Done

File: src/main/java/org/samples/todos/repository/TaskRepository.java [hunk 4]
```diff
@@ -1,23 +1,46 @@
...
 
-        //TODO: @Lucka please implement me
-        // Example: use jackson library https://www.baeldung.com/jackson-object-mapper-tutorial
+        ObjectMapper mapper = new ObjectMapper();
+
+        try {
+            taskGroups = mapper.readValue(Paths.get(TASKS_FILE_NAME).toFile(), new TypeReference<List<TaskGroup>>() {
+            });
+        } catch (IOException e) {
+                e.printStackTrace();
```
[the remark this answers is not in the document]
Comment (LucyQuantAnalyst) [reply src/main/java/org/samples/todos/repository/TaskRepository.java:30]: Done

File: src/main/java/org/samples/todos/repository/TaskRepository.java [hunk 5]
```diff
@@ -1,23 +1,46 @@
...
+        TaskRepository.TASKS_FILE_NAME = TASKS_FILE_NAME;
+    }
+
     public List<TaskGroup> load() {
-        List<TaskGroup> taskGroups = null;
+        List<TaskGroup> taskGroups =  new ArrayList<>();
 
-        //TODO: @Lucka please implement me
-        // Example: use jackson library https://www.baeldung.com/jackson-object-mapper-tutorial
+        ObjectMapper mapper = new ObjectMapper();
```
[the remark this answers is not in the document]
Comment (LucyQuantAnalyst) [reply src/main/java/org/samples/todos/repository/TaskRepository.java:24]: Done

File: src/main/java/org/samples/todos/repository/TaskRepository.java [hunk 6]
```diff
@@ -1,23 +1,46 @@
...
+import java.nio.file.Paths;
+import java.util.ArrayList;
 import java.util.List;
 
 public class TaskRepository {
 
+    private static String TASKS_FILE_NAME = "src/main/resources/tasks.json";
+
+    public TaskRepository(String TASKS_FILE_NAME) {
+        TaskRepository.TASKS_FILE_NAME = TASKS_FILE_NAME;
```
[the remark this answers is not in the document]
Comment (LucyQuantAnalyst) [reply src/main/java/org/samples/todos/repository/TaskRepository.java:18]: Done

**Pull request merged**